runtime/src/main/java/io/novafoundation/nova/runtime/multiNetwork/ChainRegistry.kt

```diff
@@ -320,3 +329,12 @@ suspend fun ChainRegistry.findEvmChainFromHexId(evmChainIdHex: String): Chain? {
 
     return findEvmChain(addressPrefix)
 }
+
+fun ChainRegistry.enabledChains() = currentChains
+    .filterList { it.isEnabled }
+    .inBackground()
+    .shareIn(this, SharingStarted.Eagerly, replay = 1)
```

Comment:
I dont think we should use sharing here:

1. The sharing is done in `ChainRegistry` scope which is basically global one. So you will leak each flow you request this way and all internal information (chains map)
2. This is quite an unexpected behavior from the **function**. Usually we done sharing for flows stored in the class fields. Sharing of a flow returned from a function call wont make any difference unless you explicitly assign it to a variabler and resuse multiple times. Which is not how you use it (as I can see)